firmware-support/bittide-sys/src/debug_register.rs

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2023 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::callisto::ReframingState;
+
+/// This is a wrapper type around the `debugRegisterWb` component. The fields are as
+/// described:
+/// 0. Reframing state (`ReframingState`):
+///   - `0x00`: Discriminant
+///   - `0x04`: Target correction (`f32`)
+///   - `0x08`: Target count (`u32`)
+#[derive(Clone)]
+pub struct DebugRegister {
+    base_addr: *const u32,
+}
+
+impl DebugRegister {
+    /// Instantiate a debug registere from a base address.
+    ///
+    /// # Safety
+    ///
+    /// `base_addr` must be a valid memory address to a debug register.
+    pub unsafe fn from_base_addr(base_addr: *const u32) -> Self {
+        DebugRegister { base_addr }
+    }
```

Comment:
Just calling it `new` is consistent with the other modules

Reply:
`from_base_addr` is also consistent with `bittide_sys::clock_control::ClockControl`. It's a personal association, but I prefer `from_base_addr` over `new` since the former has a very strict association with a pointer as an input, like `std::slice::from_raw_parts` and company. I prefer to use `new` for safe wrappers around constructors like this one.